lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -1008,3 +1008,11 @@ RequirementRequest::evaluate(Evaluator &evaluator,
   }
   llvm_unreachable("unhandled kind");
 }
+
+llvm::Expected<Type>
+swift::StructuralTypeRequest::evaluate(Evaluator &evaluator,
+                                       TypeAliasDecl *D) const {
+  auto typeRepr = D->getUnderlyingTypeLoc().getTypeRepr();
+  auto resolution = TypeResolution::forStructural(D);
+  return resolution.resolveType(typeRepr, None);
```

Comment:
You need to use the same options as in `validateTypeAliasType()`:

```
  TypeResolutionOptions options(
    (typeAlias->getGenericParams() ?
     TypeResolverContext::GenericTypeAliasDecl :
     TypeResolverContext::TypeAliasDecl));

  if (!typeAlias->getDeclContext()->isCascadingContextForLookup(
        /*functionsAreNonCascading*/true)) {
     options |= TypeResolutionFlags::KnownNonCascadingDependency;
  }
```

For now you can just copy and paste that code, we'll factor it out later.

Reply:
... however a generic type alias will never end up here, so you remove that part. In fact, can you `assert(!getGenericParams())` in `TypeAliasDecl::getStructuralType()`?